app/lib/clients/vault/oidc.rb

```diff
@@ -0,0 +1,154 @@
+=begin
+
+The purpose of this module is to assign a policy to an OIDC user, by
+mapping that user's email address to a policy we create.
+It works as follows:
+
+It creates an OIDC provider and user.  That user has a
+username/password/email addr, that can be accessed with OIDC auth.
+
+It creates an OIDC client which connects to that provider.  When a
+user tries to auth, the client connects to the provider, which opens
+up a browser window allowing the user to enter his username/password.
+
+On success, the provider returns an OIDC token, which includes the
+user's email addr.
+
+The client has been configured to map that email address to an entity
+in vault, which has the policy which we want the user to have.
+
+So the mapping goes from the email address on the provider, to the
+policy in vault.
+
+Note that this provider is only meant to be used in our dev/test
+environment to excercise the client.  In a prod env, a real OIDC
+provider is configured in config/astral.yml
+
+=end
+module Clients
+  class Vault
+    module Oidc
+      def configure_oidc_provider
```

Comment:
maybe we should separate the "configure as provider" and "configure as client" as distinct module files? just for conceptual clarity

Reply:
or adding "as" in the method names -- in this case we are not configuring Vault's oidc provider but setting up vault as a provider; in the other methods we *are* configuring Vault's oidc provider ("_as_client")?

Reply:
> maybe we should separate the "configure as provider" and "configure as client" as distinct module files? just for conceptual clarity

I think that is a good idea.  I'll separate out the modules